clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1637,6 +1637,18 @@ static std::string eraseAnonNamespace(std::string S) {
   return S;
 }
 
+static bool checkEnumTemplateParameter(const EnumDecl *ED,
+                                       DiagnosticsEngine &Diag,
+                                       SourceLocation KernelLocation) {
+  if (!ED->isScoped() && !ED->isFixed()) {
+    Diag.Report(KernelLocation, diag::err_sycl_kernel_incorrectly_named) << 2;
+    Diag.Report(ED->getSourceRange().getBegin(), diag::note_entity_declared_at)
+        << ED;
+    return true;
+  }
+  return false;
```

Comment:
Returning logic seems a bit reversed comparing with usual clang style. I got used to see `return false` when error occurs.

Reply:
Clang 'check' Functions almost always are 'return false' on success.  That is why the 'DiagnosticBuilder' object converts to bool as 'always true'.  You can do "return Diag(...)".

Reply:
Okay, that makes sense. I was confused by something wrong, then.

Reply:
Admittedly the CFE is consistently inconsistent on it and reviewers don't notice it often enough.